pyproject.toml

```diff
@@ -39,6 +39,7 @@ dependencies = [              # F39 / PyPI
     "requests>=2.25.1",       # 2.28.2 / 2.31.0
     "ruamel.yaml>=0.16.6",    # 0.17.32 / 0.17.32
     "urllib3>=1.26.5, <3.0",  # 1.26.16 / 2.0.4
+    "jira"
```

Comment:
Could we make it a separate package from the beginning and put the dependency there?

Reply:
Could you elaborate please? How would that look?

Reply:
Write it as a separate project. It can be in a different repo like what I'm working on with [tmt-cmake](https://github.com/LecrisUT/tmt-cmake) or it can live within the current git repo, e.g. [hatch](https://github.com/pypa/hatch/blob/master/pyproject.toml) contains [hatchling](https://github.com/pypa/hatch/blob/master/backend/pyproject.toml) as a sub-project.

w.r.t. how to add functionality to `tmt`, you can use [`project.entry-points."tmt.plugin"`](https://github.com/LecrisUT/tmt-cmake/blob/684a9a90618bc8fc00a9368c4a41bbfd33d8b192/pyproject.toml#L59-L62).

Should note that we still haven't pinned down how to make it available for testing-farm [^1], but let's investigate it when it's closer to production.

Side-note, this feature requires [teemtee/web](https://github.com/teemtee/web) doesn't it? It would be nice to incorporate these 2 together. An abstract `link` interface should be available as @happz [mentioned](https://github.com/teemtee/tmt/pull/2922#discussion_r1600229944) so this might be a bit more complicated to have it in a completely separate repo right now (not impossible, I do that in [tmt-cmake](https://github.com/LecrisUT/tmt-cmake/blob/684a9a90618bc8fc00a9368c4a41bbfd33d8b192/.github/workflows/step_test.yaml#L35-L37))

[^1]: https://github.com/teemtee/tmt/issues/2758#issuecomment-1995473532

Reply:
> Could you elaborate please? How would that look?

I meant to get inspiration from 'test-convert' or 'export-polarion' in `[project.optional-dependencies]`

Reply:
Fixed in f6e67b61.